internal/http_handler.go

```
@@ -87,19 +87,19 @@ func (h *HttpServer) startHTTPServer() {
 
 			//TODO Error handling
 			if req.LeaderCommit > h.s.commitIndex {
+				h.s.mu.Lock()
 				for i := h.s.commitIndex + 1; i <= req.LeaderCommit; i++ {
 					var l Log
 					data, _ := h.s.log.Read(uint64(i))
 					json.Unmarshal(data, &l)
```

golangci-lint (1.62.2)

Error return value of `json.Unmarshal` is not checked

(errcheck)